src/util.c

```diff
@@ -69,6 +69,9 @@ CK_RV p11prov_fetch_attributes(P11PROV_CTX *ctx, P11PROV_SESSION *session,
     } else if (attrnums > 1
                && (ret == CKR_ATTRIBUTE_SENSITIVE
                    || ret == CKR_ATTRIBUTE_TYPE_INVALID)) {
+        /* The previous PKCS#11 call failed, which populated the OpenSSL error queue.
+         * But we working around this error so just flush the error now. */
+        ERR_clear_error();
```

Comment:
See the other comment for error stack management

Reply:
We are hitting the issue here in the corner case for softokn, where we ask for one attribute `CKA_ALLOWED_MECHANISMS`, so we do not fall back to the fetching of attributes per partes and stack resetting on non-required attributes:
```
56: C_GetAttributeValue^M
P:683169; T:0x139804941566592 2024-10-23 15:45:49.074^M
[in] hSession = 0x1000004^M
[in] hObject = 0x9a81bced^M
[in] pTemplate[1]: ^M     
    CKA_ALLOWED_MECHANISMS  0000000000000000 / 0^M
[out] pTemplate[1]: ^M    
    CKA_ALLOWED_MECHANISMS  0000000000000000 / -1^M
Returned:  18 CKR_ATTRIBUTE_TYPE_INVALID^M
```
Not sure if current solution is best, but it also works.

Reply:
This is so annoying,
maybe we should have a quirk to turn off using CKA_ALLOWED_MECHANISM and conditionalize the template ?

Reply:
Actually we already have a quirk, so we should just use it here to decide whether we want to add CKA_ALLOWED_MECHANISMS to the template, see p11prov_rsapss_gen() for an example.

Reply:
We have just the trial-error quirk `token_supports_allowed_mechs` in `p11prov_rsapss_gen()` which does not extend outside of this function context. I will probably introduce a new configuration and turn this off for softokn altogether.
For the case in signature.c we can not use trial-error as it is the only attribute we have in the template.

Reply:
oh, the quirk is implemented, but not documented, which confused me.